src/main/java/com/google/devtools/build/lib/analysis/test/TestStrategy.java

```diff
@@ -165,10 +165,7 @@ public final boolean isTestKeepGoing() {
   public static ImmutableList<String> expandedArgsFromAction(TestRunnerAction testAction)
       throws CommandLineExpansionException {
     List<String> args = Lists.newArrayList();
-    // TODO(ulfjack): `executedOnWindows` is incorrect for remote execution, where we need to
-    // consider the target configuration, not the machine Bazel happens to run on. Change this to
-    // something like: testAction.getConfiguration().getTargetOS() == OS.WINDOWS
-    final boolean executedOnWindows = (OS.getCurrent() == OS.WINDOWS);
+    final boolean executedOnWindows = testAction.getConfiguration().getOS() == OS.WINDOWS;
```

Comment:
Actions also have access to their execution platform via [`ActionAnalysisMetadata.getExecutionPlatform()`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/actions/ActionAnalysisMetadata.java;drc=49c9475aa7db252c19157ee059c40862a412df5d;bpv=1;bpt=1;l=233?gsn=getExecutionPlatform&gs=kythe%3A%2F%2Fgithub.com%2Fbazelbuild%2Fbazel%3Flang%3Djava%3Fpath%3Dcom.google.devtools.build.lib.actions.ActionAnalysisMetadata%23c6439d067f3aa5d6f841e899332022befabb9829f2f85fb28d571c7b2402034f).